mlrun/artifacts/model.py

```diff
@@ -506,10 +504,22 @@ def _get_extra(target, extra_data, is_dir=False):
     return extra_dataitems
 
 
-def _remove_tag_from_spec_yaml(model_spec):
-    spec_dict = model_spec.to_dict()
-    spec_dict["metadata"].pop("tag", None)
-    return yaml.safe_dump(spec_dict)
+def _sanitize_model_spec_yaml(model: ModelArtifact):
+    model_dict = model.to_dict()
+
+    # The model spec yaml should not include the tag, as the same model can be used with different tags,
+    # and the tag is not part of the model spec but the metadata of the model artifact
+    model_dict["metadata"].pop("tag", None)
+
+    # Remove future packaging links
+    if model_dict["spec"].get("extra_data"):
+        model_dict["spec"]["extra_data"] = {
+            key: item
+            for key, item in model_dict["spec"]["extra_data"].items()
+            if item is not ...
+        }
+
+    return yaml.safe_dump(model_dict)
```

Comment:
this function performs 2 things
1. it sanitizes some information
2. serializes the object

I suggest to break the function to 2 and have the former ut coveraged.